smithy-aws-iam-traits/src/main/java/software/amazon/smithy/aws/iam/traits/ConditionKeysValidator.java

```diff
@@ -44,23 +47,59 @@ public final class ConditionKeysValidator extends AbstractValidator {
     public List<ValidationEvent> validate(Model model) {
         ConditionKeysIndex conditionIndex = ConditionKeysIndex.of(model);
         TopDownIndex topDownIndex = TopDownIndex.of(model);
+        OperationIndex operationIndex = OperationIndex.of(model);
 
         return model.shapes(ServiceShape.class)
                 .filter(service -> service.hasTrait(ServiceTrait.class))
                 .flatMap(service -> {
                     List<ValidationEvent> results = new ArrayList<>();
                     Set<String> knownKeys = conditionIndex.getDefinedConditionKeys(service).keySet();
 
+                    if (service.hasTrait(ConditionKeysResolvedByServiceTrait.class)) {
+                        Optional<ConditionKeysResolvedByServiceTrait> resolvedByServiceTrait =
+                                service.getTrait(ConditionKeysResolvedByServiceTrait.class);
+                        if (resolvedByServiceTrait.isPresent()) {
+                            List<String> values = resolvedByServiceTrait.get().getValues();
+                            for (String name : values) {
+                                if (!knownKeys.contains(name)) {
+                                    results.add(error(service, String.format(
```

Comment:
You can provide the `resolvedByServiceTrait` as the second parameter to `error()` to provide more accurate source locations for this specific issue.